arch/cpu/nrf/nrf5340/tz-target-cfg.c

```diff
@@ -0,0 +1,410 @@
+  NVIC_ClearTargetState(NRFX_IRQ_NUMBER_GET(NRF_UARTE0));
+#endif
+
+#ifdef SECURE_UART1
+  /* UARTE1 is a secure peripheral, so its IRQ has to target S state */
+  NVIC_ClearTargetState(NRFX_IRQ_NUMBER_GET(NRF_UARTE1));
+#endif
+
+  return TFM_PLAT_ERR_SUCCESS;
+}
+/******************************************************************************/
+/*----------------- NVIC interrupt enabling for S peripherals ----------------*/
+enum tfm_plat_err_t
+nvic_interrupt_enable(void)
+{
+  /* SPU interrupt enabling */
+  spu_enable_interrupts();
+
+  NVIC_ClearPendingIRQ(NRFX_IRQ_NUMBER_GET(NRF_SPU));
+  NVIC_EnableIRQ(NRFX_IRQ_NUMBER_GET(NRF_SPU));
+
+  return TFM_PLAT_ERR_SUCCESS;
+}
+/******************************************************************************/
+/*------------------- SAU/IDAU configuration functions -----------------------*/
+void
+sau_and_idau_cfg(void)
+{
+  /* IDAU (SPU) is always enabled. SAU is non-existent.
+   * Allow SPU to have precedence over (non-existing) ARMv8-M SAU.
+   */
+  TZ_SAU_Disable();
+  SAU->CTRL |= SAU_CTRL_ALLNS_Msk;
+}
+/******************************************************************************/
+enum tfm_plat_err_t
+spu_periph_init_cfg(void)
+{
+  /* LOG_DBG("PERIPHERAL START \n"); */
+  /* Peripheral configuration */
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_FPU));
+  spu_peripheral_config_non_secure((uint32_t)NRF_FPU, false);
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_REGULATORS));
+  spu_peripheral_config_non_secure((uint32_t)NRF_REGULATORS, false);
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_CLOCK));
+  spu_peripheral_config_non_secure((uint32_t)NRF_CLOCK, true);   /*necesary*/
+
+#ifndef SECURE_UART0
+  /* If UART0 is a secure peripheral, we need to leave Serial-Box 0 as Secure */
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_SPIM0));
+  spu_peripheral_config_non_secure((uint32_t)NRF_SPIM0, false);
+#endif
+
+#ifndef SECURE_UART1
+  /* If UART1 is a secure peripheral, we need to leave Serial-Box 1 as Secure */
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_SPIM1));
+  spu_peripheral_config_non_secure((uint32_t)NRF_SPIM1, false);
+#endif
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_SPIM4));
+  spu_peripheral_config_non_secure((uint32_t)NRF_SPIM4, false);
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_SPIM2));
+  spu_peripheral_config_non_secure((uint32_t)NRF_SPIM2, false);
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_SPIM3));
+  spu_peripheral_config_non_secure((uint32_t)NRF_SPIM3, false);
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_SAADC));
+  spu_peripheral_config_non_secure((uint32_t)NRF_SAADC, false);
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_TIMER0));
+  spu_peripheral_config_non_secure((uint32_t)NRF_TIMER0, false);
+
+  /* NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_TIMER1)); */
+  /* spu_peripheral_config_non_secure((uint32_t)NRF_TIMER1, false); */
+
+  NVIC_DisableIRQ(NRFX_IRQ_NUMBER_GET(NRF_TIMER2));
+  spu_peripheral_config_non_secure((uint32_t)NRF_TIMER2, false);
+
+  /* Before commented. */
```

Comment:
Upstream comment, or was the next statements commented out in the upstream source?